unlock-app/src/middlewares/web3Middleware.js

```diff
@@ -64,14 +39,9 @@ export default function web3Middleware({ getState, dispatch }) {
    */
   web3Service.on('lock.saved', (lock, address) => {
     dispatch(lockDeployed(lock, address))
-    dispatch(
-      updateTransaction(lock.transaction, {
-        lock: address,
-      })
-    )
     web3Service.refreshAccountBalance(getState().account)
     web3Service.getLock(address)
-    web3Service.getPastLockTransactions(address)
+    web3Service.getPastLockTransactions(address) // This is costly and not useful for the paywall app...
```

Comment:
just to check - if we have price changes and the like, is the lock retrieved by `lock.saved` the most current state for the lock?

Reply:
Yes, `getLock` will always get the most up to date version of the lock... but there may be flickering. This is actually something that I've been thinking about about earlier and forgot to open as an issue. We should probably keep track of an indication of how up to date a given object is.

Reply:
👍